src/core/manager.c

```diff
@@ -3713,9 +3713,7 @@ int manager_reload(Manager *m) {
         m->uid_refs = hashmap_free(m->uid_refs);
         m->gid_refs = hashmap_free(m->gid_refs);
 
-        r = lookup_paths_init_or_warn(&m->lookup_paths, m->runtime_scope, 0, NULL);
-        if (r < 0)
-                return r;
+        assert_se(lookup_paths_init_or_warn(&m->lookup_paths, m->runtime_scope, 0, NULL) >= 0);
```

Comment:
An assert here makes me a bit nervous, as this reads from the env vars... isn't there anything else that can be done?

Reply:
Hmm, the env of manager shouldn't change for the whole runtime, right? If the initial setup in manager_startup() succeeded but fails here, I think the system is totally borked...

Reply:
How about to call this earlier on a temporal object, then take it here?
```c
_cleanup_(lookup_paths_done) LookupPaths lookup_paths = {};
r = lookup_paths_init_or_warn(&m->lookup_paths, m->runtime_scope, 0, NULL);
if (r < 0)
        return r;

/* 💀 This is the point of no return, from here on there is no way back. 💀 */
reloading = NULL;

...

m->lookup_paths = TAKE_STRUCT(lookup_paths);
```

`assert()` or `assert_se()` should be used for checking programming error. And the failure here should not.